cliquet/resource.py

```diff
@@ -239,7 +238,8 @@ class BaseResource(object):
     mapping = ResourceSchema()
     """Schema to validate records."""
 
-    def __init__(self, request):
+    def __init__(self, request, context=None):
```

Comment:
I haven't explained it in more details here, but in case a context factory was used, the context will be populated. That means we can have a look at anything that's in the context.

In our case, to avoid double lookups for the data in the database (when we had to check the database for information, for instance in case of a PUT request) it might make sense to store the record in the context and then check the context rather than querying the database if possible.

I let this as an optimisation, for later.